crates/metal-codegen-llvm/src/core.rs

```diff
@@ -191,22 +201,21 @@ mod tests {
     fn test_nested_module_function() {
         // NOTE `children` is not appended to cause Rust sucks.
         let module = get_empty_module("core".to_string(), None);
-        let another_module = get_empty_module("lib".to_string(), Some(&module));
-        let mut last_module = get_empty_module("str".to_string(), Some(&another_module));
+        let another_module = get_empty_module("lib".to_string(), Some(module.clone()));
+        let mut last_module = get_empty_module("str".to_string(), Some(another_module.clone()));
```

Comment:
why all the duplication? `module` and `another_module` also don't seem to be used anywhere else

Reply:
> why all the duplication? `module` and `another_module` also don't seem to be used anywhere else

This is used to test module naming since a function define would need to go up the parent tree to find the full module name.

Reply:
Still, since `module` and `another_module` are not used elsewhere, it seems pointless to clone them